tools/circt-test/circt-test-runner-circt-bmc.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+from pathlib import Path
+import argparse
+import shlex
+import subprocess
+import sys
+import tempfile
+
+parser = argparse.ArgumentParser()
+parser.add_argument("mlir")
+parser.add_argument("-t", "--test", required=True)
+parser.add_argument("-d", "--directory", required=True)
+parser.add_argument("-m", "--mode", type=int)
+parser.add_argument("-k", "--depth")
+args = parser.parse_args()
+
+if args.mode is not None and args.mode != "bmc":
+  sys.stderr.write("error: circt-bmc only supports `bmc` mode")
+
+# Use circt-opt to lower any `verif.formal` ops to `hw.module`s. Once circt-bmc
+# natively supports `verif.formal`, we can directly run it on the input MLIR.
+lowered_mlir = Path(args.directory) / "lowered.mlir"
+cmd = ["circt-opt", args.mlir, "-o", lowered_mlir, "--lower-formal-to-hw"]
+sys.stderr.write("# " + shlex.join(str(c) for c in cmd) + "\n")
+sys.stderr.flush()
+result = subprocess.call(cmd)
+if result != 0:
+  sys.exit(result)
+
+# Run circt-bmc. We currently have to capture the output and look for a specific
+# string to know if the verification passed or failed. Once the tool properly
+# exits on test failure, we can skip this.
+cmd = ["circt-bmc", lowered_mlir]
+cmd += ["-b", str(args.depth or 20)]
+cmd += ["--module", args.test]
+cmd += ["--shared-libs", "/usr/lib/libz3.so"]
```

Comment:
Maybe get this path as an argument instead of hardcoding it?

Reply:
That would be nice. I'm not sure how to do that exactly, though. The idea was to prevent the user from having to configure the runners. It must be possible to type `circt-test input.mlir` and just have all the tests in there run. So maybe this should be an environment variable, or part of a local configuration file somehow? Or maybe the runner script should just try a few common locations for Z3 and otherwise report that circt-bmc is not available. But this should definitely be configurable somehow, but maybe just not as a command line argument on `circt-test`.

Reply:
Environment variable sounds even better if that's possible.

Reply:
Done! Turned this into an environment variable. Funny how integration tests flush out missing features and find issues 😬